### PR TITLE
Exclude action events from correlation results

### LINKED ISSUE #7164: Exclude underlying events from correlation with actions
## Bug description

When running a Correlation Analysis using actions, you always (or almost always) see the underlying events that comprise the action(s) with large correlation likelihoods. This is obviously pretty useless as we're using the same event to say there's a high correlation likelihood.

## Expected behavior

If I use an action in a funnel, no underlying events for that action should show up in the Correlation analysis results.

## How to reproduce

1. Create a funnel using actions as steps.
2. See underlying events pop up in Correlation analysis with huge correlation coefficients.

## Environment

- [X] PostHog Cloud
- [X] self-hosted PostHog (ClickHouse-based), version/commit: `1.30.0`
- [ ] self-hosted PostHog (Postgres-based, legacy), version/commit: _please provide_

### PR DESCRIPTION
**Pull request by neilkakkar**
## Changes

Fixes https://github.com/PostHog/posthog/issues/7164

## How did you test this code?

Unit test

### CODE REVIEW

File: ee/clickhouse/queries/funnels/test/test_funnel_correlation.py [hunk 1]
```diff
@@ -160,6 +171,104 @@ def test_basic_funnel_correlation_with_events(self):
         self.assertEqual(len(self._get_people_for_event(filter, "negatively_related", success=False)), 5)
         self.assertEqual(len(self._get_people_for_event(filter, "negatively_related")), 0)
 
+    @snapshot_clickhouse_queries
+    def test_funnel_correlation_with_events_and_actions(self):
```
Comment (hazzadous) [inline ee/clickhouse/queries/funnels/test/test_funnel_correlation.py:175]: non-blocking

If this is explicitly checking that actions are excluded, perhaps something like `test_action_events_are_excluded_from_correlations`?

File: ee/clickhouse/queries/funnels/test/test_funnel_correlation.py [hunk 2]
```diff
@@ -160,6 +171,104 @@ def test_basic_funnel_correlation_with_events(self):
...
+
+        for i in range(10):
```
Comment (hazzadous) [inline ee/clickhouse/queries/funnels/test/test_funnel_correlation.py:177]: Is there a reason for so many events? If not, I would suggest we go with the minimum reproducible test case for this specific test.

File: ee/clickhouse/queries/funnels/test/test_funnel_correlation.py [hunk 3]
```diff
@@ -160,6 +171,104 @@ def test_basic_funnel_correlation_with_events(self):
...
+            _create_person(distinct_ids=[f"user_{i}"], team_id=self.team.pk)
+            _create_event(
```
Comment (hazzadous) [inline ee/clickhouse/queries/funnels/test/test_funnel_correlation.py:179]: non-blocking as is consistent with other tests in the file.

Can we use the `ee.clickhouse.test.test_journeys.journeys_for` to construct these scenarios?

**Review by hazzadous — APPROVED** [review]
I think the solution is good, I would love to see a more concise test for the functionality, but happy not to block on that.

File: ee/clickhouse/queries/funnels/test/test_funnel_correlation.py (hunk 2 above)
Comment (neilkakkar) [reply ee/clickhouse/queries/funnels/test/test_funnel_correlation.py:177]: Fair, will update!

**Pull request merged**